daemon/algod/api/server/v2/handlers.go

```diff
@@ -965,6 +965,79 @@ func (v2 *Handlers) SimulateTransaction(ctx echo.Context, params model.SimulateT
 	if err != nil {
 		return badRequest(ctx, err, err.Error(), v2.Log)
 	}
+	request := PreEncodedSimulateRequest{
+		TxnGroups: []PreEncodedSimulateRequestTransactionGroup{
+			{
+				Txns: txgroup,
+			},
+		},
+	}
+	return v2.simulate(ctx, request, (*string)(params.Format))
+}
+
+// PreEncodedSimulateRequestTransactionGroup mirrors model.SimulateRequestTransactionGroup
+type PreEncodedSimulateRequestTransactionGroup struct {
+	model.SimulateRequestTransactionGroup
+	Txns []transactions.SignedTxn `codec:"txns"`
+}
+
+// PreEncodedSimulateRequest mirrors model.SimulateRequest
+type PreEncodedSimulateRequest struct {
+	model.SimulateRequest
+	TxnGroups []PreEncodedSimulateRequestTransactionGroup `codec:"txn-groups"`
+}
+
+// SimulateTransactionExtended simulates broadcasting a raw transaction to the network, returning relevant simulation results.
+// (POST /v2/transactions/simulate/extended)
+func (v2 *Handlers) SimulateTransactionExtended(ctx echo.Context, params model.SimulateTransactionExtendedParams) error {
+	stat, err := v2.Node.Status()
+	if err != nil {
+		return internalError(ctx, err, errFailedRetrievingNodeStatus, v2.Log)
+	}
+	if stat.Catchpoint != "" {
+		// node is currently catching up to the requested catchpoint.
+		return serviceUnavailable(ctx, fmt.Errorf("ExtendedSimulateTransaction failed as the node was catchpoint catchuping"), errOperationNotAvailableDuringCatchup, v2.Log)
+	}
+	proto := config.Consensus[stat.LastVersion]
+
+	req := ctx.Request()
+	buf := new(bytes.Buffer)
+	req.Body = http.MaxBytesReader(nil, req.Body, MaxTealDryrunBytes)
+	_, err = buf.ReadFrom(ctx.Request().Body)
+	if err != nil {
+		return badRequest(ctx, err, err.Error(), v2.Log)
+	}
+	data := buf.Bytes()
+
+	var simulateRequest PreEncodedSimulateRequest
+	err = decode(protocol.JSONStrictHandle, data, &simulateRequest)
+	if err != nil {
+		err = decode(protocol.CodecHandle, data, &simulateRequest)
+		if err != nil {
+			return badRequest(ctx, err, err.Error(), v2.Log)
+		}
+	}
+
+	for _, txgroup := range simulateRequest.TxnGroups {
+		if len(txgroup.Txns) == 0 {
+			err = errors.New("empty txgroup")
+			return badRequest(ctx, err, err.Error(), v2.Log)
+		}
+		if len(txgroup.Txns) > proto.MaxTxGroupSize {
+			err = fmt.Errorf("transaction group size %d exceeds protocol max %d", len(txgroup.Txns), proto.MaxTxGroupSize)
+			return badRequest(ctx, err, err.Error(), v2.Log)
+		}
```

Comment:
was wondering if we are worrying about failing early? for I checked the `ledger/eval/eval.go` implementation: https://github.com/algorand/go-algorand/blob/master/ledger/eval/eval.go#L936-L946, and it seems we might need to only worry about empty transaction group case, but the exceeding max txn group size case can be delegated later?